lib/galaxy/model/__init__.py

```diff
@@ -11569,6 +11569,73 @@ def __repr__(self):
         )
 
 
+class UserCredentials(Base):
+    """
+    Represents a credential associated with a user for a specific service.
+    """
+
+    __tablename__ = "user_credentials"
+
+    id: Mapped[int] = mapped_column(primary_key=True)
+    user_id: Mapped[int] = mapped_column(ForeignKey("galaxy_user.id"), index=True, nullable=False)
+    reference: Mapped[str] = mapped_column(nullable=False)
+    source_type: Mapped[str] = mapped_column(nullable=False)
+    source_id: Mapped[str] = mapped_column(nullable=False)
+    current_group_id: Mapped[int] = mapped_column(
+        ForeignKey("user_credentials_group.id", ondelete="CASCADE"), index=True, nullable=True
+    )
+    create_time: Mapped[Optional[datetime]] = mapped_column(default=now)
```

Comment:
`create_time` and `update_time` don't have to be optional: you are providing default values; besides, a record has always a create and update time. Same for other tables.